### ISSUE TITLE
Error in Zoo attack

### ISSUE THREAD

**Issue by ririnicolae** [Trusted-AI/adversarial-robustness-toolbox#60]
**Describe the bug**
Shape error in Zoo attack:
```
x_adv = attack.generate(x_test)
 File "/job/model-code/art/attacks/zoo.py", line 180, in generate
   res = self._generate_batch(x_batch, y_batch)
 File "/job/model-code/art/attacks/zoo.py", line 223, in _generate_batch
   best_dist, best_label, best_attack = self._generate_bss(x_batch, y_batch, c)
 File "/job/model-code/art/attacks/zoo.py", line 320, in _generate_bss
   x_adv = self._optimizer(x_adv, y_batch, c)
 File "/job/model-code/art/attacks/zoo.py", line 380, in _optimizer
   _,_ , loss = self._loss(expanded_x, expanded_x + coord_batch.reshape(expanded_x.shape), expanded_targets,
ValueError: cannot reshape array of size 8388608 into shape (256,3,32,32)
```
**To Reproduce**
CIFAR-10 dataset

**Comment by ririnicolae** [comment]
Fixed in b0ecc55.

**Comment by aldahdooh** [comment]
Bug still there (ART version 1.3.1)

```
Traceback (most recent call last):
  File "generate_adv.py", line 194, in <module>
    main(args)
  File "generate_adv.py", line 175, in main
    adv_data = attack.generate(x=x_test, y=y_test)
  File "/home/aaldahdo/adv_dnn/venv/lib/python3.8/site-packages/art/attacks/attack.py", line 74, in replacement_function
    return fdict[func_name](self, *args, **kwargs)
  File "/home/aaldahdo/adv_dnn/venv/lib/python3.8/site-packages/art/attacks/evasion/zoo.py", line 222, in generate
    res = self._generate_batch(x_batch, y_batch)
  File "/home/aaldahdo/adv_dnn/venv/lib/python3.8/site-packages/art/attacks/evasion/zoo.py", line 263, in _generate_batch
    best_dist, best_label, best_attack = self._generate_bss(x_batch, y_batch, c_current)
  File "/home/aaldahdo/adv_dnn/venv/lib/python3.8/site-packages/art/attacks/evasion/zoo.py", line 380, in _generate_bss
    x_adv = self._optimizer(x_adv, y_batch, c_batch)
  File "/home/aaldahdo/adv_dnn/venv/lib/python3.8/site-packages/art/attacks/evasion/zoo.py", line 455, in _optimizer
    expanded_x, expanded_x + coord_batch.reshape(expanded_x.shape), expanded_targets, expanded_c,
ValueError: cannot reshape array of size 25165824 into shape (4096,32,32,3)
```

**Comment by beat-buesser** [comment]
Hi @aldahdooh, thank you for your message. I was not able to reproduce your observation with branch `dev_1.4.3` based on example get_started_pytorch.py` change to use `ZooAttack`. Could you please provide more details or code example how to reproduce the error?

**Comment by aldahdooh** [comment]
For CIFAR-10 dataset
Note: No error with MNIST
```
attack = ZooAttack(classifier=kclassifier, batch_size=32)
adv_data = attack.generate(x=x_test, y=y_test)
```

**Comment by beat-buesser** [comment]
Hi @aldahdooh I have tried to reproduce this issue with `dev_1.3.1` and `dev_1.4.3` using `get_started_keras.py` and have not been able to reproduce the issue with the attack definition above for the first 32 test samples. Would you be able to show the entire code example producing this issue?

**Comment by aldahdooh** [comment]
[zoo_test.zip](https://github.com/Trusted-AI/adversarial-robustness-toolbox/files/5583641/zoo_test.zip)

Thanks for your reply, Attached is the code. Just run `generate_adv.py -d=cifar`. I run it with MNIST and no errors encountered. With CIFAR, the problem came with the last batch size iteration.

**Comment by beat-buesser** [comment]
Oh, I see, it happens in the last batch iteration, thank you, that's very helpful. Let me try again using your code.

**Comment by beat-buesser** [comment]
Hi @aldahdooh I have pushed an update for ZOO to #755 that should fix the issue. If you have time, could you please try it out and let me know if it works for you?

I have noticed that you are creating the attacks in `generate_Adv.py` with the attack's default arguments which are selected to let the attack run fast but not necessarily very strong.

**Comment by aldahdooh** [comment]
Hi @beat-buesser, Many thanks for your feedback. I will do that ASAP since my machine is busy now with other tasks. 
And thanks for your note, I do agree with you, I did that to compare with other works that use same settings.
